app/screens/home/search/results/file_results.tsx

```diff
@@ -0,0 +1,214 @@
+import {useTheme} from '@context/theme';
+import {useIsTablet} from '@hooks/device';
+import {useImageAttachments} from '@hooks/files';
+import {bottomSheet, dismissBottomSheet} from '@screens/navigation';
+import NavigationStore from '@store/navigation_store';
+import {isImage, isVideo} from '@utils/file';
+import {fileToGalleryItem, openGalleryAtIndex} from '@utils/gallery';
+import {bottomSheetSnapPoint} from '@utils/helpers';
+import {getViewPortWidth} from '@utils/images';
+import {TabTypes} from '@utils/search';
+import {preventDoubleTap} from '@utils/tap';
+
+import FileOptions from './file_options';
+import {HEADER_HEIGHT} from './file_options/header';
+
+import type ChannelModel from '@typings/database/models/servers/channel';
+
+const styles = StyleSheet.create({
+    container: {
+        flex: 1,
+        marginHorizontal: 20,
+    },
+});
+
+type Props = {
+    canDownloadFiles: boolean;
+    fileChannels: ChannelModel[];
+    fileInfos: FileInfo[];
+    publicLinkEnabled: boolean;
+    scrollPaddingTop: number;
+    searchValue: string;
+}
+
+const galleryIdentifier = 'search-files-location';
+
+const FileResults = ({
+    canDownloadFiles,
+    fileChannels,
+    fileInfos,
+    publicLinkEnabled,
+    scrollPaddingTop,
+    searchValue,
+}: Props) => {
+    const theme = useTheme();
+    const isTablet = useIsTablet();
+    const insets = useSafeAreaInsets();
+    const [lastViewedIndex, setLastViewedIndex] = useState<number | undefined>(undefined);
+
+    const paddingTop = useMemo(() => ({paddingTop: scrollPaddingTop, flexGrow: 1}), [scrollPaddingTop]);
+    const {images: imageAttachments, nonImages: nonImageAttachments} = useImageAttachments(fileInfos, publicLinkEnabled);
+    const channelNames = useMemo(() => fileChannels.reduce<{[id: string]: string | undefined}>((acc, v) => {
+        acc[v.id] = v.displayName;
+        return acc;
+    }, {}), [fileChannels]);
+
+    const containerStyle = useMemo(() => {
+        return {top: fileInfos.length ? 8 : 0};
+    }, [fileInfos]);
+
+    const filesForGallery = useMemo(() => imageAttachments.concat(nonImageAttachments),
+        [imageAttachments, nonImageAttachments]);
+
+    const orderedFilesForGallery = useMemo(() => (
+        filesForGallery.sort((a: FileInfo, b: FileInfo) => {
+            return (b.create_at || 0) - (a.create_at || 0);
+        })
+    ), [filesForGallery]);
+
+    const filesForGalleryIndexes = useMemo(() => orderedFilesForGallery.reduce<{[id: string]: number | undefined}>((acc, v, idx) => {
+        if (v.id) {
+            acc[v.id] = idx;
+        }
+        return acc;
+    }, {}), [orderedFilesForGallery]);
+
+    const handlePreviewPress = useCallback(preventDoubleTap((idx: number) => {
+        const items = orderedFilesForGallery.map((f) => fileToGalleryItem(f, f.user_id));
+        openGalleryAtIndex(galleryIdentifier, idx, items);
+    }), [orderedFilesForGallery]);
+
+    const snapPoints = useMemo(() => {
```

Comment:
snapPoints could have been written inside the `handleOptionsPress` method.  then, no need for `usememo`. again the thought of having too much `useMemo`

Reply:
Perhaps, you can profile this component and see what really improves if we strategically employ `useMemo`